scripts/prepare_network.py

```diff
@@ -282,13 +336,24 @@ def set_line_nom_max(n, s_nom_max_set=np.inf, p_nom_max_set=np.inf):
         if "Co2L" in o:
             m = re.findall("[0-9]*\.?[0-9]+$", o)
             if len(m) > 0:
-                co2limit = float(m[0]) * snakemake.config["electricity"]["co2base"]
-                add_co2limit(n, co2limit, Nyears)
-                logger.info("Setting CO2 limit according to wildcard value.")
+                if snakemake.config["electricity"]["automatic_emission_base_year"]:
+                    filename = download_emission_data()
+                    co2limit = emission_extractor(filename)
+                    co2limit = [i * float(m[0]) for i in co2limit]
```

Comment:
emission_extractor() function returns a list containing emission values. I have tried to multiply the list with float(m[0]) and it threw an error. After searching I was able to do it as in the code. The below picture contains two list for WAS (Western Asia Countries) emission values. In the first list co2 emissions are not multiplied, in the second list, it is multiplied by 0.20 (it can be anything below 1) and printed.

![image](https://github.com/pypsa-meets-earth/pypsa-earth/assets/62134151/77a6ccc8-32bc-45e3-a98e-63abda0e4239)

![image](https://github.com/pypsa-meets-earth/pypsa-earth/assets/62134151/c1b207f4-5726-48a8-8feb-6994fb3c78a4)

Reply:
A clean a perfect way to multiple lists :)

However, if `emission_extractor()` returns a list with elements corresponding to countries, summation seems to be lost in `emission_extractor()`. We need to transfer a single float value into `add_co2limit()`, not a list. Could you please double-check it?